rosa_tutorials/rosa-mobb-configure-cluster-pull-images-aws-ecr-tutorial.adoc

```diff
@@ -0,0 +1,175 @@
+
+Quick introduction by Ryan Niksch and Charlotte Fung on link:https://youtu.be/1PBFtpCIMBo[YouTube].
+
+There are two options to use to authenticate with AWS Elastic Container Registry (ECR) to pull images.
+
+The traditional method is to create a pull secret for ECR:
+
+.Example
+[source,terminal]
+----
+$ oc create secret docker-registry ecr-pull-secret  \
+  --docker-server=<registry id>.dkr.ecr.<region>.amazonaws.com  \
+  --docker-username=AWS --docker-password=$(aws ecr get-login-password) \
+  --namespace=hello-world
+----
+
+However Amazon ECR tokens expire every 12 hours which means you need to re-authenticate every 12 hours either through scripting or do so manually.
+
+A second, and preferred method, is to attach an ECR policy to your cluster's worker machine profiles which this guide will walk you through.
+
+== Attach ECR policy role
+
+You can attach an ECR policy to your cluster, giving the cluster permissions to pull images from your registries. ROSA worker machine instances include pre-defined IAM roles, named differently depending on whether the cluster is an STS cluster or a non-STS cluster.
+
+=== STS cluster role
+
+`ManagedOpenShift-Worker-Role` is the IAM role attached to Red Hat OpenShift Service on AWS (ROSA) STS compute instances.
+
+=== Non-STS cluster role
+
+`<cluster name>-<identifier>-worker-role` is the IAM role attached to ROSA non-STS compute instances.
+[TIP]
+====
+To find the non-STS cluster role, run the following command with the name of your cluster:
+[source,terminal]
+----
+$ aws iam list-roles | grep <cluster_name>
+----
+====
+
+image::mobb-nonsts-roles.png[Non-STS roles]
+
+== Configure ECR with ROSA
+
+ECR has several pre-defined policies that give permissions to interact with the service. In the case of ROSA, you will be pulling images from ECR and will only need to add the `AmazonEC2ContainerRegistryReadOnly` policy.
+
+. Add the `AmazonEC2ContainerRegistryReadOnly` policy to the `ManagedOpenShift-Worker-Role` for STS clusters (or the `<cluster name>-<identifier>-worker-role` for non-STS clusters).
++
+.STS Example:
++
+[source,terminal]
+----
+$ aws iam attach-role-policy \
+     --role-name ManagedOpenShift-Worker-Role \
+     --policy-arn "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly"
+----
+. Set ENV variables. Set the AWS region and registry name for creating a new ECR:
++
+[source,terminal]
+----
+$  REGION=us-east-2
+   REGISTRY=hello-ecr
+----
+. Create a repository:
++
+[source,terminal]
+----
+$ aws ecr create-repository \
+    --repository-name $REGISTRY \
+    --image-scanning-configuration scanOnPush=true \
+    --region $REGION
+----
+. Set the registry ID:
++
+[source,terminal]
+----
+$ REGISTRYID=`aws ecr describe-repositories --repository-name hello-ecr | jq -r '.repositories[].registryId'`
+----
+. Log into ECR:
++
+[source,terminal]
+----
+$ podman login -u AWS -p $(aws ecr get-login-password --region $REGION) $REGISTRYID.dkr.ecr.$REGION.amazonaws.com
```

Comment:
To run this command, podman must start a machine, otherwise will get errors for command run